skimage/util/_backends.py

```diff
@@ -0,0 +1,153 @@
+        public_name = "skimage.filters.rank"
+    else:
+        public_name = ".".join(full_name.split(".")[:2])
+
+    # It would be nice to sanity check things by doing something like the
+    # following. However we can't because this code is executed while the
+    # module is being imported, which means this would create a circular
+    # import
+    # mod = importlib.import_module(public_name)
+    # assert getattr(mod, func.__name__) is func
+
+    return public_name
+
+
+@cache
+def all_backends():
+    """List all installed backends and information about them"""
+    backends = {}
+    backends_ = _entry_points(group="skimage_backends")
+    backend_infos = _entry_points(group="skimage_backend_infos")
+
+    for backend in backends_:
+        backends[backend.name] = {"implementation": backend}
+        try:
+            info = backend_infos[backend.name]
+            # Double () to load and then call the backend information function
+            backends[backend.name]["info"] = info.load()()
+        except KeyError:
+            pass
+
+    return backends
+
+
+def dispatchable(func):
+    """Mark a function as dispatchable.
+
+    When a decorated function is called the installed backends are
+    searched for an implementation. If no backend implements the function
+    then the scikit-image implementation is used.
+    """
+    func_name = func.__name__
+    func_module = get_module_name(func)
+
+    # If no backends are installed at all or dispatching is disabled,
+    # return the original function. This way people who don't care about it
+    # don't see anything related to dispatching
+    if disable_dispatching() or not all_backends():
+        return func
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        # Backends are tried in alphabetical order, this makes things
+        # predictable and stable across runs. Might need a better solution
+        # when it becomes common that users have more than one backend
+        # that would accept a call.
+        for name in sorted(all_backends()):
+            backend = all_backends()[name]
+            # If we have a BackendInformation object we check if the
+            # function we are looking for is implemented in the backend
+            if "info" in backend:
+                if (
+                    f"{func_module}:{func_name}"
+                    not in backend["info"].supported_functions
+                ):
+                    continue
+
+            backend_impl = backend["implementation"].load()
+
+            # Allow the backend to accept/reject a call based on the function
+            # name and the values of the arguments
+            wants_it = backend_impl.can_has(
+                f"{func_module}:{func_name}", *args, **kwargs
+            )
+            if not wants_it:
+                continue
+
+            # can_has("foo", ...) might be True, but the function might not actually
+            # be implemented in the backend
```

Comment:
Could you explain why we should support this case, `can_has` is true but the backend doesn't implement the function? Nothing comes to my mind?

Reply:
Good point, most likely it is some kind of leftover from previous iterations. Removed.